### ISSUE TITLE
🐛 Bug Report: Setting `auth.enableExperimentalRedirectFlow` still results in popup window upon login

### ISSUE THREAD

**Issue by jslott2sigma** [backstage/backstage#19852]
### 📜 Description

I set the following in my config (and confirmed via devtools):
```
auth:
    enableExperimentalRedirectFlow: true
```

I log out and log back in, however, I still see a popup when logging in with the Azure provider. Here is `App.tsx`:
```
        SignInPage: props => {
            const discoveryApi = useApi(discoveryApiRef);
            return (
                <SignInPage
                    {...props}
                    auto
                    provider={{
                        id: 'microsoft',
                        title: 'Azure',
                        message: 'Sign in using Azure',
                        apiRef: microsoftAuthApiRef,
                    }}
                    onSignInSuccess={async (identityApi: IdentityApi) => {
                        setTokenCookie(await discoveryApi.getBaseUrl('cookie'), identityApi);
                        props.onSignInSuccess(identityApi);
                    }}
                />
            );
        },
```
Is there anything else I need to do to enable redirect (vs popup) auth flow? Or suggestions on how to debug?

### 👍 Expected behavior

Upon login, page should redirect rather than popup

### 👎 Actual Behavior with Screenshots

It's a bit tough to capture a screenshot of the login dialog that appears and disappears quite quickly unfortunately.

### 👟 Reproduction steps

1. Set `auth.enableExperimentalRedirectFlow` to true in config
2. Restart Backstage
3. Visit Backstage at port 3000, on Settings page, Logout
4. Note that a popup appears upon re-login

### 🖥️ Your Environment

OS: Linux 5.4.245-ts1-amd64 - linux/x64
node: v18.14.2
backstage: 1.17.5
Dependencies:
    @backstage/app-defaults: 1.4.2
    @backstage/backend-app-api: 0.5.2
    @backstage/backend-common: 0.19.4
    @backstage/backend-dev-utils: 0.1.1
    @backstage/backend-openapi-utils: 0.0.3
    @backstage/backend-plugin-api: 0.6.2
    @backstage/backend-tasks: 0.5.7
    @backstage/catalog-client: 1.4.3
    @backstage/catalog-model: 1.4.1
    @backstage/cli-common: 0.1.12
    @backstage/cli-node: 0.1.3
    @backstage/cli: 0.22.12
    @backstage/config-loader: 1.4.0
    @backstage/config: 1.0.8
    @backstage/core-app-api: 1.9.1
    @backstage/core-components: 0.13.4
    @backstage/core-plugin-api: 1.5.3
    @backstage/dev-utils: 1.0.20
    @backstage/errors: 1.2.1
    @backstage/eslint-plugin: 0.1.3
    @backstage/integration-aws-node: 0.1.5
    @backstage/integration-react: 1.1.18
    @backstage/integration: 1.6.2
    @backstage/plugin-adr-backend: 0.3.8
    @backstage/plugin-adr-common: 0.2.14
    @backstage/plugin-adr: 0.6.6
    @backstage/plugin-api-docs: 0.9.10
    @backstage/plugin-app-backend: 0.3.50
    @backstage/plugin-app-node: 0.1.2
    @backstage/plugin-auth-backend: 0.18.8
    @backstage/plugin-auth-node: 0.2.19
    @backstage/plugin-catalog-backend-module-github: 0.3.7
    @backstage/plugin-catalog-backend-module-gitlab: 0.2.6
    @backstage/plugin-catalog-backend-module-unprocessed: 0.2.2
    @backstage/plugin-catalog-backend: 1.12.4
    @backstage/plugin-catalog-common: 1.0.15
    @backstage/plugin-catalog-graph: 0.2.35
    @backstage/plugin-catalog-import: 0.9.13
    @backstage/plugin-catalog-node: 1.4.3
    @backstage/plugin-catalog-react: 1.8.3
    @backstage/plugin-catalog-unprocessed-entities: 0.1.2
    @backstage/plugin-catalog: 1.12.4
    @backstage/plugin-devtools-backend: 0.1.5
    @backstage/plugin-devtools-common: 0.1.3
    @backstage/plugin-devtools: 0.1.3
    @backstage/plugin-events-node: 0.2.11
    @backstage/plugin-github-actions: 0.6.4
    @backstage/plugin-home-react: 0.1.2
    @backstage/plugin-home: 0.5.7
    @backstage/plugin-org: 0.6.13
    @backstage/plugin-permission-common: 0.7.7
    @backstage/plugin-permission-node: 0.7.13
    @backstage/plugin-permission-react: 0.4.14
    @backstage/plugin-proxy-backend: 0.3.2
    @backstage/plugin-scaffolder-backend: 1.16.5
    @backstage/plugin-scaffolder-common: 1.4.0
    @backstage/plugin-scaffolder-node: 0.2.2
    @backstage/plugin-scaffolder-react: 1.5.4
    @backstage/plugin-scaffolder: 1.14.4
    @backstage/plugin-search-backend-module-catalog: 0.1.6
    @backstage/plugin-search-backend-module-pg: 0.5.11
    @backstage/plugin-search-backend-module-techdocs: 0.1.6
    @backstage/plugin-search-backend-node: 1.2.6
    @backstage/plugin-search-backend: 1.4.2
    @backstage/plugin-search-common: 1.2.5
    @backstage/plugin-search-react: 1.6.4
    @backstage/plugin-search: 1.3.6
    @backstage/plugin-tech-insights-backend-module-jsonfc: 0.1.34
    @backstage/plugin-tech-insights-backend: 0.5.16
    @backstage/plugin-tech-insights-common: 0.2.11
    @backstage/plugin-tech-insights-node: 0.4.8
    @backstage/plugin-tech-insights: 0.3.15
    @backstage/plugin-tech-radar: 0.6.7
    @backstage/plugin-techdocs-backend: 1.6.7
    @backstage/plugin-techdocs-module-addons-contrib: 1.0.18
    @backstage/plugin-techdocs-node: 1.7.6
    @backstage/plugin-techdocs-react: 1.1.9
    @backstage/plugin-techdocs: 1.6.8
    @backstage/plugin-user-settings: 0.7.8
    @backstage/release-manifests: 0.0.9
    @backstage/test-utils: 1.4.2
    @backstage/theme: 0.4.1
    @backstage/types: 1.1.0
    @backstage/version-bridge: 1.0.4
    @types/react-dom: 18.2.7
    @types/react-is: 18.2.0, 18.2.1
    @types/react-redux: 7.1.25
    @types/react-sparklines: 1.7.2
    @types/react-swipeable-views-utils: 0.13.4
    @types/react-swipeable-views: 0.13.2
    @types/react-text-truncate: 0.14.1
    @types/react-transition-group: 4.4.6
    @types/react: 18.2.20
    react-base16-styling: 0.6.0
    react-beautiful-dnd: 13.1.1
    react-copy-to-clipboard: 5.1.0
    react-debounce-input: 3.3.0
    react-dev-utils: 12.0.1
    react-dom: 18.2.0
    react-double-scrollbar: 0.0.15
    react-draggable: 4.4.5
    react-error-overlay: 6.0.11
    react-event-listener: 0.6.6
    react-fast-compare: 3.2.2
    react-grid-layout: 1.3.4
    react-helmet: 6.1.0
    react-hook-form: 7.43.9, 7.44.3
    react-hookz: 1.0.12
    react-immutable-proptypes: 2.2.0
    react-immutable-pure-component: 2.2.2
    react-inspector: 6.0.2
    react-is: 16.13.1, 17.0.2, 18.2.0
    react-json-view: 1.21.3
    react-lifecycles-compat: 3.0.4
    react-markdown: 8.0.7
    react-redux: 7.2.9, 8.1.0
    react-refresh: 0.14.0
    react-resizable: 3.0.5
    react-router-dom: 6.13.0
    react-router: 6.13.0
    react-side-effect: 2.1.2
    react-sparklines: 1.7.0
    react-swipeable-views-core: 0.14.0
    react-swipeable-views-utils: 0.14.0
    react-swipeable-views: 0.14.0
    react-syntax-highlighter: 15.5.0
    react-text-truncate: 0.19.0
    react-textarea-autosize: 8.5.2
    react-transition-group: 4.4.5
    react-universal-interface: 0.6.2
    react-use: 17.4.0
    react-virtualized-auto-sizer: 1.0.20
    react-window: 1.8.9
    react: 18.2.0

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by awanlin** [comment]
Hi @jslott2sigma, we are using this feature and it works properly for us. One "gotcha" we ran into is that the config needs to be at the root and not under `auth:`, can you try that, please?

Code for reference: https://github.com/backstage/backstage/blob/044e8146f5ebc3ee8758027893edfb499651926d/packages/core-app-api/config.d.ts#L124

**Comment by jslott2sigma** [comment]
Yeah that was it, thanks @awanlin! I guess I got confused when reading the changelog here: https://github.com/backstage/backstage/blob/master/plugins/auth-backend/CHANGELOG.md.

**Comment by awanlin** [comment]
Yup, I did the same. I'll try and put a PR together to update that so others don't run into as well

**Comment by jslott2sigma** [comment]
Cool! Has this feature migrated beyond the `experimental` designation too? Is your plan to add an actual `auth.enableRedirectFlow` and deprecate `enableExperimentalRedirectFlow`? (I'd even argue is should default=true at some point!) ;)

Thanks again for the prompt responses.

**Comment by awanlin** [comment]
That's I'm less sure about to be honest

@Rugvip, any chance you happen to know when this might move away from being "experimental"?

**Comment by awanlin** [comment]
Just noting that I have a PR submitted to improve the docs around this feature: https://github.com/backstage/backstage/pull/19891
